ui/app/js/controllers/required_signoffs_controller.js

```diff
@@ -0,0 +1,371 @@
+          "signoffs_required": rs.signoffs_required || 0,
+          "sc_id": rs.sc_id,
+          "scheduled_by": rs.scheduled_by,
+          "sc_data_version": rs.sc_data_version,
+          "signoffs": rs.signoffs,
+          "change_type": rs.change_type,
+        };
+      });
+    }
+  })
+  .error(function(response) {
+    sweetAlert(
+      "Failed to load pending Product Required Signoffs:",
+      response
+    );
+  })
+  .finally(function() {
+    loading_deferreds["product_sc"].resolve();
+  });
+
+  PermissionsRequiredSignoffs.getScheduledChanges()
+  .success(function(response) {
+    if (response["count"] > 0) {
+      response["scheduled_changes"].forEach(function(rs) {
+        if (! (rs.product in $scope.required_signoffs)) {
+          $scope.required_signoffs[rs.product] = {};
+        }
+
+        if (! ("permissions" in $scope.required_signoffs[rs.product])) {
+          $scope.required_signoffs[rs.product]["permissions"] = {};
+        }
+
+        if (! (rs.role in $scope.required_signoffs[rs.product]["permissions"])) {
+          $scope.required_signoffs[rs.product]["permissions"][rs.role] = {
+            "signoffs_required": 0,
+            "data_version": null,
+          };
+        }
+
+        $scope.required_signoffs[rs.product]["permissions"][rs.role]["sc"] = {
+          "required_signoffs": rs.required_signoffs,
+          "signoffs_required": rs.signoffs_required || 0,
+          "sc_id": rs.sc_id,
+          "scheduled_by": rs.scheduled_by,
+          "sc_data_version": rs.sc_data_version,
+          "signoffs": rs.signoffs,
+          "change_type": rs.change_type,
+        };
+      });
+    }
+  })
+  .error(function(response) {
+    sweetAlert(
+      "Failed to load pending Permissions Required Signoffs:",
+      response
+    );
+  })
+  .finally(function() {
+    loading_deferreds["permissions_sc"].resolve();
+  });
+
+  Permissions.getCurrentUser()
+  .success(function(response) {
+    $scope.current_user = response["username"];
+    $scope.user_roles = Object.keys(response["roles"]);
+  })
+  .error(function(response) {
+    sweetAlert(
+      "Failed to load current user Roles:",
+      response
+    );
+  })
+  .finally(function() {
+    loading_deferreds["current_user"].resolve();
+  });
+
+
+  $scope.addNewRequiredSignoff = function() {
+    $modal.open({
+      templateUrl: "required_signoff_modal.html",
+      controller: "NewRequiredSignoffCtrl",
+      backdrop: "static",
+      resolve: {
+        required_signoffs: function() {
+          return $scope.required_signoffs;
+        },
+        current_user: function() {
+          return $scope.current_user;
+        },
+      }
+    });
+  };
+
+  $scope.editRequiredSignoffs = function(mode, channel) {
+    channel = channel ? channel : "";
+    $modal.open({
+      templateUrl: "required_signoff_modal.html",
+      controller: "EditRequiredSignoffsCtrl",
+      backdrop: "static",
+      resolve: {
+        mode: function() {
```

Comment:
I mean we can now use a lambda-like function in JS. They are called [arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions). This will make the code just a little bit smaller:
```js
mode: () => mode,
channel: () => channel,
```